lambda/src/app.js

```diff
@@ -111,3 +107,67 @@ exports.lambdaHandler = async (event, context) => {
     };
   }
 };
+
+async function rekognizeLabels(imageBytes) {
+  const params = {
+    Image: {
+      Bytes: imageBytes,
+    },
+    MaxLabels: 3,
+    MinConfidence: 80,
+  };
+
+  return rekognition.detectLabels(params).promise();
+}
+
+async function uploadImageToS3(bucket, key, buffer) {
+  const params = {
+    Bucket: bucket,
+    Key: key,
+    Body: buffer,
+    ContentEncoding: "base64",
+    ContentType: "image",
+  };
+  return s3.putObject(params).promise();
+}
+
+async function insertIntoDynamo(event, id, pet, imagePath) {
+  const item = {
+    id: { S: id },
+    pet: { S: pet },
+    imagePath: { S: imagePath },
+    userAgent: { S: event.headers["User-Agent"] },
+    desktopClient: {
+      BOOL: event.headers["CloudFront-Is-Desktop-Viewer"] === true,
+    },
+    mobileClient: {
+      BOOL: event.headers["CloudFront-Is-Mobile-Viewer"] === true,
+    },
+    smartTvClient: {
+      BOOL: event.headers["CloudFront-Is-SmartTV-Viewer"] === true,
+    },
+    tabletClient: {
+      BOOL: event.headers["CloudFront-Is-Tablet-Viewer"] === true,
+    },
+    country: { S: event.headers["CloudFront-Viewer-Country"] },
+    // TODO any other useful info that can be gleaned?
+  };
+
+  removeNullFields(item);
+  console.log(item);
+
+  return db.putItem({ TableName: tableName, Item: item }).promise();
+}
+
+function getFileType(encodedFile) {
+  return encodedFile.split("data:image/").pop().split(";base64,")[0];
+}
+
+function getImageBuffer(encodedFile) {
+  const base64Image = encodedFile.split("base64,")[1];
+  return Buffer.from(base64Image, "base64");
+}
+
+function getS3ObjectPath(bucket, key, region) {
```

Comment:
This already has access to the region, not sure why passing it in. Could just have the entire URL prefix as the constant. This function doesn't add much.